sdk/core/azure-core/inc/azure/core/http/win_http_transport.hpp

```diff
@@ -134,6 +134,27 @@ namespace Azure { namespace Core { namespace Http {
     bool IgnoreUnknownCertificateAuthority = false;
   };
 
+  namespace _internal {
+    /**
+     * @brief Factory to produce a context with special settings like no client cert context.
+     *
+     * @details Client SDKs can use this class to create an Azure Context to change the
+     * WinHttpTransport default behavior. This is because a Client SDK doesn't know about the
+     * specifics of a transport adapter. So, on Windows, SDK client can set options and only
+     * WinHTTPTransport will check for those options to be in the context.
+     *
+     */
+    class WinHttpTransportContextProvider {
+    public:
+      // Factory with no constructor
+      WinHttpTransportContextProvider() = delete;
+
+      // Creates a set up token to make
+      static Azure::Core::Context GetNoClientCertificateContext(Azure::Core::Context const& parent);
```

Comment:
Can I use this context for other winhttp configuration purposes? Or is this only for setting the no client cert option?

Reply:
Yes and no.
A Context is a list of linked nodes where each node can hold only one key/value configuration.
When calling this function, a node is added to the context passed as parameter.  
People can later create and add more nodes to the context by creating more children nodes and hence, setting more options.